step1/src/main.js

```diff
@@ -1,20 +1,224 @@
 /**
- * AppenderForm에서 submit 이벤트가 발생했을 때 처리하는 함수
- * @param event
+ * todoList의 아이템들을 관리하기 위한 객체
  */
-function handleSubmitAppenderForm (event) {
-  const input = event.target.querySelector('input');
-  alert(`input에 입력된 텍스트: ${input.value}`);
-  alert(`input에 입력된 텍스트의 길이: ${input.value.length}`);
+const todoItems = {
+  items: [],              // ex) { id: 1, content: 'todoList만들기', highlight: false, isComplete: false, createdAt: Date.now()}    
+  seletedItem: -1,        // 특정 아이템 체크 여부를 확인하기 위한 변수
+};
+
+/**
+ * 템플릿을 제공하는 함수 
+ */
+const template = function() {
+  return `
+  <h1>📃 TodoList</h1>
+    <form name="appenderForm" action="" method="post">
+      <fieldset>
+        <legend hidden>TodoList Form</legend>
+        <label>
+          <span hidden>아이템 추가</span>
+          <input type="text" size="40" placeholder="Todo Item 내용을 입력해주세요">
+        </label>
+        <button type="submit">전송</button>
+      </fieldset>
+    </form>
+    <ul>
+      ${todoItems.items.map(function(item) {
+        if(todoItems.seletedItem == item.id) {
+          return `
+            <li>
+              <form name="modifierForm" action="" id="${item.id}">
+                <fieldset>
+                  <legend hidden>아이템 수정</legend>
+                  <label>
+                    <span hidden>아이템 수정</span>
+                    <input type="text" value="${item.content}" size="40">
+                  </label>
+                  <button type="submit">완료</button>
+                  <button type="button" class="cancel">취소</button>
+                </fieldset>
+              </form>
+            </li>
+          `
+      }
+      return `
+        <li>
+          <p ${item.highlight ? 'style="color:#09F"' : ''} ${item.isComplete ? 'class="finish"' : ''}>
+            <input type="checkbox" id="${item.id}" ${item.highlight ? ' checked' : ''}>
+            ${item.content}
+          </p>
+          <button type="button" class="complete" id="${item.id}">완료</button>
+          <button type="button" class="modify" id="${item.id}">수정</button>
+          <button type="button" class="delete" id="${item.id}">삭제</button>
+        </li>
+      `
+    }).join('')}
+    </ul>
+  `
+};
```

Comment:
앞서 언급한 것 처럼 `itemTemplate` 함수를 만들어줄 경우, 다음과 같이 template을 표현할 수 있겠죠?
```suggestion
const template = () => `
  <h1>📃 TodoList</h1>
  <form name="appenderForm" action="" method="post">
    <fieldset>
      <legend hidden>TodoList Form</legend>
      <label>
        <span hidden>아이템 추가</span>
        <input type="text" size="40" placeholder="Todo Item 내용을 입력해주세요">
      </label>
      <button type="submit">전송</button>
    </fieldset>
  </form>
  <ul>
      ${todoItems.items.map(itemTemplate).join('')}
  </ul>
`;
```

이런식으로 코드를 분리하고 줄여나갈 수 있답니다!